main.go

```diff
@@ -63,35 +65,27 @@ func run(conf *sat.Config) error {
 	// we don't collect this error.
 	serverErrors := make(chan error, 1)
 
+	// start sat and its internal reactr/vektor/grav
 	go func() {
-		localLogger.Info("startup", "sat started")
 		serverErrors <- s.Start()
 	}()
 
-	// start scanning for our procfile being deleted
+	// create and scan for our procfile
 	go func() {
-		for {
-			if _, err = process.Find(conf.ProcUUID); err != nil {
-				localLogger.Warn("proc file deleted, sending termination signal")
-				shutdown <- syscall.SIGTERM
-				break
-			}
-
-			time.Sleep(time.Second)
-		}
+		serverErrors <- createAndScanProcFile(log, conf, shutdown)
```

Comment:
this muddies the channels. If `create` fails, func returns error, gets sent to the `serverErrors` channel here

if procfile is deleted, signal gets sent to the passed in shutdown signal within the function. These two look lopsided.

What do you think about this pseudocode construction instead:

```go
go func() {
    if err := createProfile(conf); err != nil {
        serverErrors <- err
    }

    scanProfile(conf, log, shutdown)
}

func createProfile(conf) error {
    // write a file to disk which describes this instance
	info := process.NewInfo(conf.Port, conf.JobType)
	if err := info.Write(conf.ProcUUID); err != nil {
		return errors.Wrap(err, "failed to Write process info")
	}

    return nil
}

func scanProfile(conf, log, shutdown) {
    for {
		if _, err := process.Find(conf.ProcUUID); err != nil {
			log.Warn("proc file deleted, sending termination signal")
			shutdown <- syscall.SIGTERM
			break
		}

		time.Sleep(time.Second)
	}
}
```

It's a bit more code, but the two are separated, and clearly defines the expectations and makes reasoning about the individual pieces easier.